interfaces/Portalicious/src/app/services/auth.service.ts

```diff
@@ -113,12 +87,16 @@ export class AuthService {
   }
 
   public async login(
-    credentials: { username: string; password?: string },
+    credentials?: { username: string; password?: string },
     returnUrl?: string,
   ) {
     this.logService.logEvent(LogEvent.userLogin);
+    if (returnUrl) {
+      setReturnUrlInLocalStorage(returnUrl);
+    }
     const user = await this.authStrategy.login(credentials);
-    this.setUserInStorage(user);
+    // Note: SSO never resolves so the code below this line is not executed in the SSO case
+    setUserInLocalStorage(user);
```

Comment:
```suggestion
    // Note: SSO never resolves so the code below this line is not executed in the SSO case
    setUserInLocalStorage(user);
    
    if (!!this.authStrategy.CallbackComponent) {
      return this.router.navigate([AppRoutes.authCallback]);
    }
```

It's a small addition, but it would make it a bit more explicit.

Also, we could still alternatively _always_ redirect to the callback, and make it mandatory. And then implement a callback for the basic-auth strategy that only does the redirect and nothing else. I think this might be cleaner.